src/Element.cc

```diff
@@ -141,6 +141,17 @@ void Element::AddValue(const std::string &_type,
       _type, _defaultValue, _required, _description);
 }
 
+/////////////////////////////////////////////////
+void Element::AddValue(const std::string &_type,
+                       const std::string &_defaultValue,
+                       bool _required,
+                       sdf::Errors &_errors,
+                       const std::string &_description)
+{
+  this->dataPtr->value = this->CreateParam(this->dataPtr->name,
```

Comment:
Just realized that errors are sometimes using `SDF_ASSERT`  and others the `sdferr` stream. The first one throws an exception while the later prints using `sdf::Console`. Since these are two different behaviors I guess we can do either one of these:
* we don't care about changing this behavior and use either `sdferr` or `SDF_ASSERT` by default in the calling function.
* we add some "extra" info to `sdf::Errors`, maybe by creating a new enum that says `ASSERT_ERROR` or `SDFERR_...` so we can differentiate between both of them and keep the same behavior.

I also realized this might change behavior since a call to `SDF_ASSERT` will break the normal execution flow whereas if we add errors to `sdf::Errors` the function will keep executing. I think this can be solved with just a `return` and then with the throwing the `SDF_ASSERT` in the calling function but I just wanted to leave it here in case I miss something.

Reply:
Like I mentioned in another comment, I'd say let's keep the SDF_ASSERT. We can still call `AddValue` with a throwaway `Errors` object and call `SDF_ASSERT` if there are any errors.

Reply:
Yes, will try to make sure we keep the same behavior.